### PR TITLE
ER-700 Moved rules engine along with profanity rule.

### PR DESCRIPTION
**Pull request by dev4valtech**
- On review creation the review status is set to new
- Hooked up into the Review Created event to run the rules

### CODE REVIEW

File: src/Employer/UnitTests/Recruit.Vacancies.Client/Application/Rules/VacancyRules/VacancyProfanityChecksRuleTests.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Application.Rules.BaseRules;
+using Esfa.Recruit.Vacancies.Client.Application.Rules.VacancyRules;
+using Esfa.Recruit.Vacancies.Client.Application.Services;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Xunit;
+
+namespace Esfa.Recruit.Employer.UnitTests.Recruit.Vacancies.Client.Application.Rules.VacancyRules
```
Comment (sachpatel) [inline src/Employer/UnitTests/Recruit.Vacancies.Client/Application/Rules/VacancyRules/VacancyProfanityChecksRuleTests.cs]: Could you either create a more specific namespace or chop off the `Esfa.Recruit.Employer.UnitTests.` or call the namespace `Recruit.Vacancies.Client.UnitTests.Application.Rules.VacancyRules`

File: src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/CreateVacancyReviewCommandHandler.cs [hunk 2]
```diff
@@ -55,6 +55,14 @@ public async Task Handle(CreateVacancyReviewCommand message, CancellationToken c
             var vacancy = vacancyTask.Result;
             var previousReviews = previousReviewsTask.Result;
 
+            //Defensive code, just in case the message is republished due to an error after creating the review. 
+            var activePreviousReview = previousReviews.FirstOrDefault(r => r.Status != ReviewStatus.Closed);
```
Comment (leewadhams) [inline src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/CreateVacancyReviewCommandHandler.cs]: Could use .Any()

Comment (dev4valtech) [reply src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/CreateVacancyReviewCommandHandler.cs]: I am logging the id of the active review below, hence needed a reference to it.

File: src/Employer/UnitTests/Recruit.Vacancies.Client/Application/Rules/VacancyRules/VacancyProfanityChecksRuleTests.cs (hunk 1 above)
Comment (dev4valtech) [reply src/Employer/UnitTests/Recruit.Vacancies.Client/Application/Rules/VacancyRules/VacancyProfanityChecksRuleTests.cs]: Moved the class to the correct location

File: src/Employer/UnitTests/Recruit.Vacancies.Client/Application/Rules/VacancyRules/VacancyProfanityChecksRuleTests.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+        internal static Vacancy SetSkills(this Vacancy entity, IEnumerable<string> skills)
+        {
+            entity.Skills = skills.ToList();
+
+            return entity;
+        }
+    }
+
+    public class TestProfanityListProvider : IProfanityListProvider
```
[the remark this answers is not in the document]
Comment (dev4valtech) [reply src/Employer/UnitTests/Recruit.Vacancies.Client/Application/Rules/VacancyRules/VacancyProfanityChecksRuleTests.cs]: Fixed

**Pull request merged**